File: backend/CertificatesManagerApi/Mappers/SupplierMapper.cs
using CertificatesManagerApi.DTOs;
using CertificatesManagerApi.Entities;

namespace CertificatesManagerApi.Mappers
{
    public class SupplierMapper
    {
        public static SupplierDTO ToDto(Supplier supplier)
        {
            SupplierDTO supplierDTO = new()
            {
                Handle = supplier.Handle,
                Name = supplier.Name,
                Index = supplier.Index,
                City = supplier.City
            };

            return supplierDTO;
        }

        public static Supplier ToEntity(SupplierDTO supplierDTO)
        {
            Supplier supplier = new()
            {
                Handle = supplierDTO.Handle,
                Name = supplierDTO.Name,
                Index = supplierDTO.Index,
                City = supplierDTO.City
            };

            return supplier;
        }
    }
}

File: frontend/src/common/utils/mappers/supplier-mapper.ts
import { SupplierDTO } from '../../models/dtos/supplier-dto';
import Supplier from '../../models/supplier.model';

export class SupplierMapper {
	static ToModel(supplierDTO: SupplierDTO): Supplier {
		let supplier: Supplier = {
			id: supplierDTO.handle,
			name: supplierDTO.name,
			indexValue: supplierDTO.index.toString(),
			city: supplierDTO.city,
		};

		return supplier;
	}
}
